internal/image/resource_image.go

```diff
@@ -76,6 +87,18 @@ func (r ImageResource) Metadata(_ context.Context, req resource.MetadataRequest,
 func (r ImageResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
 	resp.Schema = schema.Schema{
 		Attributes: map[string]schema.Attribute{
+			"source_file": schema.SingleNestedAttribute{
+				Optional: true,
+				Attributes: map[string]schema.Attribute{
+					"data_path": schema.StringAttribute{
+						Required: true,
+					},
+					"metadata_path": schema.StringAttribute{
+						Optional: true,
+					},
```

Comment:
@stgraber After revisiting, I think both of these attributes do require a plan modifier `RequiresReplace()`, do you agree?

```suggestion
					"data_path": schema.StringAttribute{
						Required: true,
						PlanModifiers: []planmodifier.String{
							stringplanmodifier.RequiresReplace(),
						},
					},
					"metadata_path": schema.StringAttribute{
						Optional: true,
						PlanModifiers: []planmodifier.String{
							stringplanmodifier.RequiresReplace(),
						},
					},
```